### PR TITLE
[main][bugfix] Fix bugs and refactor cached mask generation logic

### PR DESCRIPTION
**Pull request by rjg-lyh**
### What this PR does / why we need it?
This PR fix bugs and refactor cached mask generation logic. Now just pre-construct and use the cached mask on cpu instead of device on npu.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
CI passed with new added/existing test.

- vLLM version: v0.10.1.1
- vLLM main: https://github.com/vllm-project/vllm/commit/9b5f64238fbd0f98928587b3426cbf69eea96ae7

### CODE REVIEW

**Review by wangxiyuan — APPROVED** [review]

File: tests/ut/attention/test_attention_mask.py [hunk 1]
```diff
@@ -77,80 +86,29 @@ def test_get_splitfuse_attn_mask(self):
...
-        self.assertEqual(attn_mask.shape, (1, 512))
-
-    def test_use_multiple_masks(self):
-        max_seq_lens = [128, 512, 1024]
-        dtypes = [torch.float16, torch.bfloat16, torch.int8]
-        for max_seq_len, dtype in zip(max_seq_lens, dtypes):
-            with self.subTest(max_seq_len=max_seq_len, dtype=dtype):
-                self._test_use_multiple_masks(max_seq_len, dtype)
-
-    def _test_use_multiple_masks(self, max_seq_len, dtype):
```
Comment (ApsarasX) [inline tests/ut/attention/test_attention_mask.py:116]: I don't agree to drop this case in which `AttentionMaskBuilder->get_splitfuse_attn_mask->get_attn_mask` with same `max_seq_len`.

The reason I added this test case in the first place was to test for mask pollution issues. Even though the refactoring has eliminated mask pollution for now, it’s still necessary to keep this test case.

**Review by ApsarasX — CHANGES_REQUESTED** [review]

File: tests/ut/attention/test_attention_mask.py (hunk 1 above)
Comment (rjg-lyh) [reply tests/ut/attention/test_attention_mask.py:116]: I think I mostly understand your point. In fact, renaming this function to 'test mask value pollution' would make it less prone to misinterpretation. I will add test cases based on the refactored version of the code.

**Review by MengqingCao — APPROVED** [review]
lgtm

**Comment by wangxiyuan** [comment]
@ApsarasX please double check the response, if it's fine. Feel free to merge this.

File: tests/ut/attention/test_attention_mask.py (hunk 1 above)
Comment (ApsarasX) [reply tests/ut/attention/test_attention_mask.py:116]: > I think I mostly understand your point. In fact, renaming this function to 'test mask value pollution' would make it less prone to misinterpretation. I will add test cases based on the refactored version of the code.

Okay, I'll approve this PR first. After adding test cases related to mask pollution, please add me as a reviewer.

**Review by ApsarasX — APPROVED** [review]

File: tests/ut/attention/test_attention_mask.py [hunk 2]
```diff
@@ -77,80 +86,48 @@ def test_get_splitfuse_attn_mask(self):
...
+        # otherwise raise ValueError
+        with self.assertRaises(ValueError):
+            attn_mask = attention_mask_builder.get_splitfuse_attn_mask(
+                seq_lens=torch.tensor([10, 20, 100]),
+                position=torch.tensor([7, 8, 9, 18, 19, 99]),
+                dtype=torch.int8,
+                device=torch.device("cpu"),
+            )
+
+    def test_mask_value_cleanliness(self):
```
Comment (rjg-lyh) [inline tests/ut/attention/test_attention_mask.py:116]: I have add this test here. @ApsarasX

Comment (ApsarasX) [reply tests/ut/attention/test_attention_mask.py:116]: > I have add this test here. @ApsarasX

OK, I see.
